Build failed in ```no_std``` environment.

Hello,

I'm building rust-mbedtls with following command and it failed.

**Version:** 
```commit f9fefe06a36986dd2c00494801c596c472bed476 (HEAD, tag: mbedtls_v0.11.0, tag: mbedtls-sys-auto_v3.5.0-alpha.2+0b3de6f, tag: mbedtls-platform-support_v0.3.0```

**Step:**
```
cargo build -p mbedtls -Zbuild-std=core,alloc,compiler_builtins --target=x86_64-unknown-none --no-default-features --features="no_std_deps,rdrand"
```

**Results:**
```
moving the metadata is recommended to avoid confusion
   Compiling mbedtls-platform-support v0.3.0 (/home/xiaoyu/rust-mbedtls/mbedtls-platform-support)
   Compiling once_cell v1.17.1
   Compiling mbedtls-sys-auto v3.5.0-alpha.2+0b3de6f (/home/xiaoyu/rust-mbedtls/mbedtls-sys)
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-unknown-none` target may not support the standard library
  = note: `std` is required by `once_cell` because it does not declare `#![no_std]`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`

error[E0463]: can't find crate for `std`
 --> /home/xiaoyu/.cargo/registry/src/github.com-1ecc6299db9ec823/once_cell-1.17.1/src/imp_std.rs:6:5
  |
6 | use std::{
  |     ^^^ can't find crate
  |
  = note: the `x86_64-unknown-none` target may not support the standard library
  = help: consider building the standard library from source with `cargo build -Zbuild-std`

error: cannot find macro `panic` in this scope
   --> /home/xiaoyu/.cargo/registry/src/github.com-1ecc6299db9ec823/once_cell-1.17.1/src/lib.rs:726:47
    |
726 |                 init.take().unwrap_or_else(|| panic!("Lazy instance has previously been poisoned"))
    |                                               ^^^^^
    |
    = note: consider importing this macro:
            core::panic

error: cannot find macro `panic` in this scope
   --> /home/xiaoyu/.cargo/registry/src/github.c
```

**Quick investigate:**

https://github.com/fortanix/rust-mbedtls/blob/f9fefe06a36986dd2c00494801c596c472bed476/mbedtls-platform-support/Cargo.toml#L25

The "std" feature of once_cell is not disabled when building the no_std target.

--- a/mbedtls-platform-support/Cargo.toml
+++ b/mbedtls-platform-support/Cargo.toml
@@ -22,7 +22,7 @@ links = "mbedtls-platform-support"
 cfg-if = "1.0.0"
 spin = { version = "0.4.0", default-features = false, optional = true }
 chrono = { version = "0.4", optional = true }
-once_cell = "1.17.1"
+once_cell = { version = "1.17.1", default-features = false }
 
 [target.x86_64-fortanix-unknown-sgx.dependencies]
 chrono = "0.4"
@@ -39,7 +39,7 @@ cc = "1.0"
 
 [features]
 time = ["mbedtls-sys-auto/time"]
-std = ["mbedtls-sys-auto/std"]
+std = ["mbedtls-sys-auto/std", "once_cell/std"]
 force_aesni_support = ["mbedtls-sys-auto/custom_has_support", "mbedtls-sys-auto/aes_alt", "aesni"]
 aesni = ["mbedtls-sys-auto/aesni"]
 tls13 = ["mbedtls-sys-auto/tls13"]

